### PR TITLE
Fall back to user provided smarts if rdkit can't find match

### PR DESCRIPTION
**Pull request by jkausrelay**
- Simple fall back for trivial mcs match

### CODE REVIEW

File: timemachine/fe/atom_mapping.py [hunk 1]
```diff
@@ -190,6 +195,10 @@ def get_core_with_alignment(
     ff: Forcefield or None
         Forcefield to use for alignment, defaults to DEFAULT_FF forcefield if None
 
+    initial_smarts: str or None
+        If set use smarts as the initial seed to MCS and as a fallback
```
Comment (badisa) [inline timemachine/fe/atom_mapping.py:199]:
```suggestion
        If set uses smarts as the initial seed to MCS and as a fallback
```

File: timemachine/fe/atom_mapping.py [hunk 2]
```diff
@@ -207,17 +216,27 @@ def get_core_with_alignment(
...
 
-    conf_a, conf_b = align_mols_by_core(mol_a, mol_b, heavy_atom_core, ff, n_steps=n_steps, k=k)
-    set_romol_conf(a_copy, conf_a)
-    set_romol_conf(b_copy, conf_b)
+        core, smarts = setup_core(a_copy, b_copy, True, initial_smarts)
+        return core, smarts
+    except AtomMappingError as err:
+        # Fall back to user provided smarts
+        if initial_smarts is not None:
+            print(f"WARNING: Could not get atom mapping: {err}, fall back to user defined smarts: {initial_smarts}")
```
Comment (badisa) [inline timemachine/fe/atom_mapping.py:237]:
```suggestion
            print(f"WARNING: Could not get atom mapping: {err}, falling back to user defined smarts: {initial_smarts}")
```

**Review by badisa — APPROVED** [review]

**Review by maxentile — APPROVED** [review]

File: timemachine/fe/atom_mapping.py [hunk 3]
```diff
@@ -46,15 +46,19 @@ def mcs(
     # atoms
     params.AtomCompareParameters.CompleteRingsOnly = 1
     params.AtomCompareParameters.RingMatchesRingOnly = 1
-    params.AtomCompareParameters.matchValences = 0
+    params.AtomCompareParameters.MatchValences = 0
```
Comment (proteneer) [inline timemachine/fe/atom_mapping.py:49]: good catch! fortunately it looks like we got lucky since the default `matchValences=False`

**Review by proteneer — APPROVED** [review]
lgtm

**Pull request merged**